docs/source/architecture/sub_stakes.rst

```diff
@@ -0,0 +1,270 @@
+			* 2nd sub-stake = 300 tokens for 7 periods
+
+    .. aafig::
+        :proportional:
+        :textual:
+
+                 Before             
+
+            stake                 
+            ^                     
+            |                     
+         900+----------+          
+            |          |  
+            |    1st   |
+            |          |  period 
+            +----------+--->   
+            0          5          
+			
+			
+			
+                 After             
+
+            stake                 
+            ^                     
+            |                     
+         900+----------+       
+            |          |
+            |    1st   |
+         300+----------+--+
+            |      2nd    |  period
+            +-------------+--->   
+            0          5  7     
+
+
+Slashing
+^^^^^^^^
+*See:* `the slashing protocol`_ 
+
+.. _`the slashing protocol`: https://docs.nucypher.com/en/latest/architecture/slashing.html
+
+
+
+
+Flags that affect the sub-stake array
+-------------------------------------
+
+Re-staking
+^^^^^^^^^^
+*Used in methods* : ``confirmActivity()``, ``mint()``
+
+When re-staking is turned off, the number of locked tokens in sub-stakes does not change by itself.
+In case when re-staking parameter is on then all staking reward is locked as the part of each involved sub-stake (inside ``confirmActivity()`` and/or ``mint()``).  Accordingly, each such sub-stake has increased locked amount (by reward) and the number of sub-stakes stays unchanged.
+
+**Example:**
+
+    A staker has few sub-stakes and calls ``mint()``:
+		- Before calling: 
+			* 1st sub-stake = 400 tokens for 10 periods
+			* 2nd sub-stake = 200 tokens for 2 periods
+			* 3rd sub-stake = 100 tokens locked starting from the next period and locked for 5 periods
+			* 100 tokens in an unlocked state
+		- After calling, re-staking is disabled:  
+			* 1st sub-stake = 400 tokens for 10 periods
+			* 2nd sub-stake = 200 tokens for 2 periods
+			* 3rd sub-stake = 100 tokens locked starting from the next period and locked for 5 periods
+			* 170 tokens in an unlocked state
+		- After calling, re-staking is enabled: 
+			* 1st sub-stake = 450 tokens for 10 periods
+			* 2nd sub-stake = 220 tokens for 2 periods
+			* 3rd sub-stake = 100 tokens locked starting from the next period and locked for 5 periods
+			* 100 tokens in an unlocked state
+
+    .. aafig::
+        :proportional:
+        :textual:
+
+                 Before             
+
+            stake
+            ^
+            |
+         700|     +----+
+            |     | 3rd|
+         600+-----+----+
+         500|          +-------------+
+            |    2nd   |     3rd     |
+         400+----------+-------------+----------+
+            |                                   |
+            |               1st                 |   period
+            +-----------------------------------+--->    
+            0     1    2             5          10
```

Comment:
Since the `3rd` sub-stake is to start at period and last for 5 periods, should the end of the `3rd` sub-stake correspond to `6` instead of `5` on the timeline? - Same for the "After" diagram below?

Reply:
I'm glad that somebody asked this :)
So thing is that locking for 5 periods means 5 future period - 1,2,3,4,5.  3rd sub-stake will last up to 5th period and will end in 6th. Current period is already part of history from this perspective.

Reply:
I tried to show discrete figure, when there is no such thing like 1->2, for example lock for one period  will be shown as vertical line. 
In all calculation current period ignored, so almost everything starts from 1 on those graphs. So lock for 10 periods will be for 1,2,3...9,10 period number, 0 is not included in this row. I guess this's problem in the explanation, maybe add "future periods" everywhere?

Reply:
How about such figure:
![image](https://user-images.githubusercontent.com/22225795/72432807-ff91cc80-37a8-11ea-929e-8abd331f2677.png)
I'm trying to use `aafigure` lib, it helps to maintain in future, but if still not good - I'll search something else

Reply:
@vzotova, 👍 I can get on board with that if that works with `aafigure`.

Reply:
Updated all diagrams 
waiting your review (also better to check version after compilation)